### PR TITLE
Command Palette: Assume current site for commands

### PR DESCRIPTION
**Pull request by rcrdortiz**
Related to:
- https://github.com/Automattic/dotcom-forge/issues/5842

## Proposed Changes

* We add an injection point to the existing code where, after the normal process, we determine if we should filter and map the siteFunction onclick to a command callback based on if we have a current site or not.

## Testing Instructions

* Use the Calypso Live Link
* Operate with the command palette with a site selected, you should see single site commands.
* Operate with the command palette without a site selected (/sites), you should see multi-site commands.
* All commands should work as expected.
* Check for regressions.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/command-palette/src/use-command-palette.tsx [hunk 1]
```diff
@@ -273,6 +283,39 @@ export const useCommandPalette = ( {
 		finalSortedCommands.unshift( viewMySitesCommand );
 	}
 
-	// Return the sorted commands
+	const currentSite = sites.find( ( site ) => site.ID === currentSiteId );
+
+	// If we have a current site and the route includes the site slug, filter and map the commands for single site use.
+	if ( currentSite && currentRoute?.includes( ':site' ) ) {
```
Comment (dsas) [inline packages/command-palette/src/use-command-palette.tsx:289]: nice!

File: packages/command-palette/src/use-command-palette.tsx [hunk 2]
```diff
@@ -273,6 +283,39 @@ export const useCommandPalette = ( {
...
+					setSearch: params.setSearch,
+					setPlaceholderOverride: params.setPlaceholderOverride,
+					command,
+				} );
+			};
+
+			return {
+				name: command.name,
+				label: command.label,
+				...( command.subLabel ? { subLabel: command.subLabel } : {} ),
```
Comment (dsas) [inline packages/command-palette/src/use-command-palette.tsx:310]: This is basically an allow list of which attributes to use. Would it be more future proof (or just more risky) to replace with a denylist of attributes not to extract?

**Review by dsas — APPROVED** [review]
All tests nicely, left one comment which is a non-blocking question/thought.

File: packages/command-palette/src/use-command-palette.tsx (hunk 2 above)
Comment (rcrdortiz) [reply packages/command-palette/src/use-command-palette.tsx:310]: I'd say it's pretty much the same. I don't think that the attribute list will grow, and we will eventually throw all this in the garbage (once we implement GB's command palette).

**Pull request merged**